modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/Classes.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.serialization.marshal;
+
+import java.io.Serializable;
+
+/**
+ * Utilities to work with classes.
+ */
+class Classes {
+    static boolean isSerializable(Class<?> objectClass) {
+        return Serializable.class.isAssignableFrom(objectClass);
+    }
+
+    static boolean isLambda(Class<?> objectClass) {
+        return objectClass.getSimpleName().contains("$$Lambda$") && objectClass.isSynthetic()
```

Comment:
I wonder if this "$$Lambda$" is JVM specific

Reply:
Yes, it's JVM specific. Actually, when checking for a lambda here, we actually want to know that the class cannot be loaded by name, so it makes sense to check this property directly.

Reworked the fix to detect whether we can load the class by name (and that it's not a primitive or an array); along with `isSynthetic()` it should yield the correct result.